src/ir_passes/await.ml

```diff
@@ -68,12 +81,14 @@ let rec t_async context exp =
    (* add the implicit return label *)
    let k_ret = fresh_cont (typ exp1) T.unit in
    let k_fail = fresh_err_cont T.unit in
+   let k_clean = fresh_cont T.unit T.unit in
    let context' =
-     LabelEnv.add Return (Cont (ContVar k_ret))
-       (LabelEnv.add Throw (Cont (ContVar k_fail)) LabelEnv.empty)
+     LabelEnv.add Cleanup (Cont (ContVar k_clean))
```

Comment:
No idea how to test for uninterrupted `Cleanup` chain for the `Fut` case. "This side" should not trap on message sends. The "other side" (i.e. the execution) is behind a context switch.